### ISSUE TITLE
sending new line character to UITextEntryLine via set_text crashes TextBoxLayoutRow (TypeError: 'NoneType' object is not subscriptable)

### ISSUE THREAD

**Issue by wadsworj** [MyreMylar/pygame_gui#273]
**Describe the bug**
When I create a UITextEntryLine and then send it a newline character via set_text pygame_gui crashes

**To Reproduce**
Steps to reproduce the behaviour:
1. create UITextEntryLine 
2. call 'set_text' method on newly created UITextEntryLine 
3. when shown pygame_gui will crash

**Expected behaviour**
I expect this to not crash.
Not sure if we want to show the '\n' or show nothing at all for that character, but I don't want it to crash.

**Stack Trace**
```
  File "C:\Users\James\AppData\Local\Programs\Python\Python310\lib\site-packages\pygame_gui\elements\ui_text_entry_line.py", line 267, in set_text
    self.drawable_shape.text_box_layout.set_cursor_position(self.edit_position)
  File "C:\Users\James\AppData\Local\Programs\Python\Python310\lib\site-packages\pygame_gui\core\text\text_box_layout.py", line 538, in set_cursor_position
    row.set_cursor_position(cursor_pos - letter_acc)
  File "C:\Users\James\AppData\Local\Programs\Python\Python310\lib\site-packages\pygame_gui\core\text\text_box_layout_row.py", line 391, in set_cursor_position
    cursor_draw_width += sum([char_metric[4]
  File "C:\Users\James\AppData\Local\Programs\Python\Python310\lib\site-packages\pygame_gui\core\text\text_box_layout_row.py", line 391, in <listcomp>
    cursor_draw_width += sum([char_metric[4]
TypeError: 'NoneType' object is not subscriptable
```

**Platform and software (please complete the following information):**
 - Windows 10
 - Pygame GUI version [e.g. 0.6.0] (how do I find this out?)
 - Pygame version [e.g. 2.0.0, 2.1.0] (how do I find this out?)

**Additional context**
The offending code:
```
                cursor_draw_width += sum([char_metric[4]
                                          for char_metric
                                          in chunk.font.get_metrics(chunk.text[:chunk_letter_pos])])
```
text_box_layout_row.py: 391

**Comment by wadsworj** [comment]
if possible I'd like to take a crack at fixing this myself

**Comment by MyreMylar** [comment]
Sure, give it a try

**Comment by wadsworj** [comment]
fixed in pull request #274
